players/player.cpp

```diff
@@ -2116,96 +2116,94 @@ int cmdVisible(const std::shared_ptr<Player>& player, cmd* cmnd) {
     return(0);
 }
 
-//********************************************************************
-//                      cmdDice
-//********************************************************************
-
 int cmdDice(const std::shared_ptr<Creature>& player, cmd* cmnd) {
-    char    *str=nullptr, *tok=nullptr, diceOutput[256], add[256];
-    int     strLen=0, i=0;
-    int     diceSides=0,diceNum=0,diceAdd=0;
-    int     rolls=0, total=0;
 
-    const char *Syntax =    "\nSyntax: dice 1d2\n"
-                            "        dice 1d2+3\n";
+    std::string input, diceSidesStr, diceNumStr, diceModStr;
 
-    strcpy(diceOutput, "");
-    strcpy(add ,"");
+    std::string syntax = "\nInvalid syntax.\ne.g: dice 5d6, dice 3d10+31, dice 4d7-3\n";
 
-    strLen = cmnd->fullstr.length();
+    int diceNum=0, diceSides=0, diceMod=0, rolls=0, total=0;
+    int maxNum = 500, maxSides=500, maxMod=10000;
 
-    // This kills all leading whitespace
-    while(i<strLen && isspace(cmnd->fullstr[i]))
-        i++;
-    // This kills the command itself
-    while(i<strLen && !isspace(cmnd->fullstr[i]))
-        i++;
-
-    str = strstr(&cmnd->fullstr[i], "d");
-    if(!str)
-        return(cmdAction(player, cmnd));
-
-    str = strdup(&cmnd->fullstr[i]);
-    if(!str) {
-        player->print(Syntax);
+    input = getFullstrText(cmnd->fullstr, 1);
+    if (input == "") {
+        for(rolls=0;rolls<2;rolls++)
+            total += Random::get(1, 6);
+        *player << "You roll 2d6: " << total << "\n";
+        broadcast(player->getSock(), player->getParent(), "(Dice 2d6): %M rolled a %d.", player.get(), total );
```

Comment:
Could use this condition to set the local vars so that it shares the same roll and output logic as the rest, if you want

Reply:
good idea